mesonbuild/interpreter/interpreter.py

```diff
@@ -258,6 +259,20 @@ class InterpreterRuleRelaxation(Enum):
     'version',
 }
 
+def _gen_deprecated_default_options() -> T.Dict[str, T.Tuple[str, str]]:
+    ret: T.Dict[str, T.Tuple[str, str]] = {}
+    for lang, envvar in compilers.compilers.CFLAGS_MAPPING.items():
```

Comment:
Hm, CFLAGS_MAPPING does not contain all languages. I get the idea that the problem is it overrides env, but for consistency I think we should still deprecate all languages, no?